src/main/java/shukaro/warptheory/handlers/warpevents/WarpWandDrain.java

```diff
@@ -21,28 +23,34 @@ public void sendChatMessage(EntityPlayer player) {
 
     @Override
     public int triggerEvent(int eventAmount, World world, EntityPlayer player) {
-        // If we try to drain more vis than the player's wand has, it'll fail, so we'll have to
-        // try to drain a little at a time.
-        int successful = 0;
-        while (successful < eventAmount) {
+        // In order to avoid this event being too punishing, here's how the trigger works:
+        //
+        // * We'll drain a small amount of vis from random aspects, up to all of them, based on the
+        //   event amount. If event amount > # of primal aspects, the excess is ignored.
+        // * If we were able to drain any vis at all, we'll clear the event entirely, to avoid the
+        //   situation where players accumulate a massive queue of wand drain events while not
+        //   holding a wand for a while.
+        boolean successful = false;
+        List<Aspect> primalAspects = Aspect.getPrimalAspects();
+        Collections.shuffle(primalAspects);
+
+        for (int i = 0; i < primalAspects.size() && i < eventAmount; i++) {
             AspectList aspectList = new AspectList();
-            for (Aspect aspect : Aspect.getPrimalAspects()) {
-                int amount = Math.max(0, world.rand.nextInt(10 * 100) - 100);
-                aspectList.add(aspect, amount);
-            }
+            int amount = Math.max(0, world.rand.nextInt(10 * 100) - 100);
+            aspectList.add(primalAspects.get(i), amount);
 
             if (ThaumcraftApiHelper.consumeVisFromInventory(player, aspectList)) {
-                successful++;
-            } else {
-                break;
+                successful = true;
```

Comment:
Just so i get it correct. This whole for function will run up to 6 times max because of how many primal aspects there are?

Reply:
Yup, exactly

Reply:
This will be so much better than before it doing every aspect every event queued up. No longer shall i have my fully changed wand be turned into a empty wand in a second :D